### PR TITLE
Fix warning (not only) in users' code

### PR DESCRIPTION
**Pull request by sideeffffect**
Get rid of
```
pattern var ex in method load is never used: use a wildcard `_` or suppress this warning with `ex@_`
@nativeLoader("adder")
 ^
```
and similar annoyances.

### CODE REVIEW

File: core/src/main/scala-2/com/github/sbt/jni/Process.scala [hunk 1]
```diff
@@ -3,8 +3,8 @@ package com.github.sbt.jni
 object Process {
   def out(command: String): String =
     try {
-      scala.sys.process.Process("uname -sm").lineStream.head
+      scala.sys.process.Process("uname -sm").!!.linesIterator.next()
```
Comment (pomadchin) [inline core/src/main/scala-2/com/github/sbt/jni/Process.scala:6]: Hm, in this case we don't need `{scala-2/scala-3}.com.github.sbt.jni.Process`. I believe this syntax works in Scala 3 as well.

Comment (pomadchin) [reply core/src/main/scala-2/com/github/sbt/jni/Process.scala]: Btw just to mentiont that explicitly: we can completely remove `com.github.sbt.jni.Process`

**Comment by sideeffffect** [comment]
Btw, why is the code from `NativeLoader` not reused in `nativeLoaderAnnotationMacro`?

**Comment by pomadchin** [comment]
@sideeffffect that's a great question, the answer is to make macro annotation a zero dependency case. To use macro annotation it's enough to have `sbt-jni-core` as a provided dependency. However, this is not possible in Scala 3, that's why I implemented this `syntax` which is close to the original ease of the annotation usage, but requires to have a dependency on the jni-core project.

File: core/src/main/scala/com/github/sbt/jni/Process.scala [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.github.sbt.jni
+
+object Process {
+  def out(command: String): String =
+    try {
+      scala.sys.process.Process("uname -sm").!!.linesIterator.next()
+    } catch {
+      case _: Exception => sys.error("Error running `uname` command")
+    }
+}
```
Comment (pomadchin) [inline core/src/main/scala/com/github/sbt/jni/Process.scala:10]: Let's get rid of this class! :D

**Comment by sideeffffect** [comment]
But the code in both seems awfully similar. How much opportunity for code reuse/sharing do you think there is?

**Comment by pomadchin** [comment]
@sideeffffect I don't think there is, we could try to think of smth in Scala 2.x I guess, but def macros also requires a dependency (I checked) 🤷 

As time goes I think we are just going to drop macro annotation / replace it with some future Scala 3 annotions (if there would be any annotations).

**Comment by sideeffffect** [comment]
OK, let's merge this.
Could you make a release after merge, so that I can use this right away, please?

**Comment by pomadchin** [comment]
@sideeffffect shipped https://github.com/sbt/sbt-jni/actions/runs/1075263499 🚀

**Pull request merged**